File: leetcode/cpp/2917_Find_the_K-or_of_an_Array.cpp
#include <vector>
using namespace std;
class Solution {
public:
	int findKOr(vector<int>& nums, int k) {
		int freq[32];
		memset(freq, 0, sizeof(freq));

		for (int& n : nums) {
			for (int b = 0; b < 32; ++b) {
				freq[b] += ((n >> b) & 1);
			}
		}

		int res = 0;
		for (int i = 0; i < 32; ++i) {
			if (freq[i] >= k) 
				res |= (1 << i);
		}

		return res;
	}
};

File: leetcode/cpp/2917_Find_the_K-or_of_an_Array.py
class Solution:
    def findKOr(self, nums: List[int], k: int) -> int:
        freq = [0] * 32

        for n in nums:
            for b in range(32):
                if (n >> b) & 1:
                    freq[b] += 1

        res = 0
        for b in range(32):
            if freq[b] >= k:
                res |= (1 << b)

        return res
